### PR TITLE
Fix CVE-2023-50711 in cloud-hypervisor

### PR DESCRIPTION
**Pull request by sindhu-karri**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
PR patches the vendor packages vmm-sys-util, vhost and versionize to fix CVE-2023-50711

###### Change Log  

- Patch CVE-2023-50711 in vendor/vmm-sys-util, vendor/vhost, vendor/versionize

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-50711

###### Test Methodology

- Pipeline build id: local build
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=486413&view=results

### CODE REVIEW

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec [hunk 1]
```diff
@@ -19,9 +19,11 @@ Source0:        https://github.com/cloud-hypervisor/cloud-hypervisor/archive/ref
 #   cd %{name}-%{version}
 #   cargo vendor > config.toml
 #   tar -czf %{name}-%{version}-cargo.tar.gz vendor/
-Source1:        %{name}-%{version}-cargo.tar.gz
+# rename the tarball to %{name}-%{version}-cargo.tar.gz when updating version
+Source1:        %{name}-%{version}-cargo-v1.tar.gz
 Source2:        config.toml
 Patch0:         CVE-2023-45853.patch
+Patch1:         CVE-2023-50711.patch
```
Comment (mfrw) [inline SPECS/cloud-hypervisor/cloud-hypervisor.spec:26]: It seems we have not used this patch.

Comment (sindhu-karri) [reply SPECS/cloud-hypervisor/cloud-hypervisor.spec:26]: Thanks Falak for noticing!

**Review by liuw — APPROVED** [review]

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec [hunk 2]
```diff
@@ -19,9 +19,15 @@ Source0:        https://github.com/cloud-hypervisor/cloud-hypervisor/archive/ref
 #   cd %{name}-%{version}
 #   cargo vendor > config.toml
 #   tar -czf %{name}-%{version}-cargo.tar.gz vendor/
+# rename the tarball to %{name}-%{version}-cargo.tar.gz when updating version
 Source1:        %{name}-%{version}-cargo.tar.gz
 Source2:        config.toml
 Patch0:         CVE-2023-45853.patch
+Patch1:         CVE-2023-50711-vmm-sys-util-fix1.patch
```
Comment (mfrw) [inline SPECS/cloud-hypervisor/cloud-hypervisor.spec:26]: NIT: [Non blocking cosmetic suggestion]

We could club `Patch1 & Patch2` in a single file & similarly club `Patch3 to Patch5` again in a single file.

**Review by mfrw — APPROVED** [review]
LGTM

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec [hunk 3]
```diff
@@ -19,9 +19,15 @@ Source0:        https://github.com/cloud-hypervisor/cloud-hypervisor/archive/ref
 #   cd %{name}-%{version}
 #   cargo vendor > config.toml
 #   tar -czf %{name}-%{version}-cargo.tar.gz vendor/
+# rename the tarball to %{name}-%{version}-cargo.tar.gz when updating version
```
[the remark this answers is not in the document]
Comment (sindhu-karri) [reply SPECS/cloud-hypervisor/cloud-hypervisor.spec:22]: Hi Tobias, We're applying patch to the source, and not modifying the source tarball. Hence we need not update the signatures file.

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec (hunk 2 above)
Comment (sindhu-karri) [reply SPECS/cloud-hypervisor/cloud-hypervisor.spec:26]: I clubbed patches 1 and 2 as they are in the same package.

**Comment by sindhu-karri** [comment]
@christopherco @romoh @jiria Please review

File: SPECS/cloud-hypervisor/cloud-hypervisor.spec (hunk 3 above)
[the remark this answers is not in the document]
Comment (tobiasb-ms) [reply SPECS/cloud-hypervisor/cloud-hypervisor.spec:22]: Huh. I must have misread. Thanks.

**Review by tobiasb-ms — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Pull request merged**